db/change_cache.go

```diff
@@ -960,3 +981,37 @@ func (l *SkippedSequenceList) getOlderThan(skippedExpiry time.Duration) []uint64
 	l.lock.RUnlock()
 	return oldSequences
 }
+
+func (c *changeCache) _getOldestPending() uint64 {
+	if c.pendingLogs.Len() > 0 {
+		return c.pendingLogs[0].Sequence
+	}
+	return 0
+}
+
+func (c *changeCache) getMaxStableCached() uint64 {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	return c._getMaxStableCached()
+}
+
+func (c *changeCache) _getMaxStableCached() uint64 {
+	oldestPending := c._getOldestPending()
+	oldestSkipped := c.getOldestSkippedSequence()
```

Comment:
I think this logic should just be 
- if getOldestSkippedSequence is non-zero, return that
- otherwise, return c.nextSequence - 1